flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/CodeGenUtils.scala

```diff
@@ -132,6 +136,15 @@ object CodeGenUtils {
    * Returns a term for representing the given class in Java code.
    */
   def typeTerm(clazz: Class[_]): String = {
+    if (clazz == classOf[StringData]) {
```

Comment:
Could this put CodeGen at risk of missing cast?
If possible, I prefer cast in accessing its methods.

Reply:
Consider `RowData` and others, using `StringData` more reasonable to me.

Reply:
There are too many calls on the `BinaryStringData` in code generation now, if we use StringData in code generation, we have to refactor a lot of codes. And we don't benefit much from this effort. I created FLINK-17437 to track this, we can refactor this in the future.